src/client/editor/nodes/ImageNode.js

```diff
@@ -0,0 +1,82 @@
+import THREE from "../../vendor/three";
+import EditorNodeMixin from "./EditorNodeMixin";
+import Image from "../objects/Image";
+
+export default class ImageNode extends EditorNodeMixin(Image) {
+  static legacyComponentName = "image";
+
+  static nodeName = "Image";
+
+  static async deserialize(editor, json) {
+    const node = await super.deserialize(editor, json);
+
+    const { src } = json.components.find(c => c.name === "image").props;
+
+    await node.load(src);
+
+    return node;
+  }
+
+  constructor(editor) {
+    super(editor);
+
+    this._canonicalUrl = null;
+  }
+
+  get src() {
+    return this._canonicalUrl;
+  }
+
+  set src(value) {
+    this.load(value).catch(console.error);
+  }
+
+  loadTexture(src) {
+    return this.editor.textureCache.get(src);
+  }
+
+  async load(src) {
+    this._canonicalUrl = src;
+    const proxiedUrl = await this.editor.project.getProxiedUrl(src);
+    return super.load(proxiedUrl);
+  }
+
+  copy(source, recursive) {
+    super.copy(source, recursive);
+
+    this._canonicalUrl = source._canonicalUrl;
+
+    return this;
+  }
+
+  serialize() {
+    const json = super.serialize();
+
+    json.components.push({
+      name: "image",
+      props: {
+        src: this._canonicalUrl
+      }
+    });
+
+    return json;
+  }
+
+  prepareForExport() {
+    const replacementObject = new THREE.Object3D().copy(this, false);
+
+    replacementObject.userData.gltfExtensions = {
+      HUBS_components: {
+        image: {
+          src: this._canonicalUrl
+        },
+        networked: {
+          id: this.uuid
+        }
+      }
+    };
+
+    this.parent.add(replacementObject);
+    this.parent.remove(this);
```

Comment:
Again probably not specific to this PR since I assume this is just a pattern we are using all around Spoke but its pretty confusing why the scene graph ends up getting manipulated during export. What is responsible for putting the editor node back in? 

Why would this not just `serialize` (or a something very similar, since I guess serialize/deserialize is for the editor format no the runtime format)? I assume this is also where all the "copy" issues you have talked about in the past comes from. Is this all just so that we don't have to modify the GLTF serialzier from Three? 

Seems like it could be done in a similar way to what we did on the loading side, by augmenting it with callbacks for extensions. Though I admittedly have not thought about it as much as you have so I may be over simplifying.

Reply:
We clone before mutating the scene graph so we don't really have to worry about the mutations. I'd rather just modify GLTFExporter so we didn't have to do this, but there are currently no hooks to be able to accomplish that. It is a complicated API to design, maybe we should just fork the GLTFExporter and figure out where we should create API hooks.

Reply:
Cloning/copying are also needed for duplicating objects. So we still need those methods to work properly.